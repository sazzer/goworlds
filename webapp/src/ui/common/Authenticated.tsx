import React, {FunctionComponent, PropsWithChildren} from 'react';
import {useSelector} from "react-redux";
import {selectCurrentUserId} from "../../authentication/currentUserId";

/**
 * Display the nested children only if we're authenticated - i.e. we have a Current User ID in the state
 * @constructor
 */
export const Authenticated: FunctionComponent<PropsWithChildren<any>> = ({children}) => {
    const currentUserId = useSelector(selectCurrentUserId);

    if (currentUserId) {
        return children;
    } else {
        return <></>;
    }
};

/**
 * Display the nested children only if we're not authenticated - i.e. we do not have a Current User ID in the state
 * @constructor
 */
export const Unauthenticated: FunctionComponent<PropsWithChildren<any>> = ({children}) => {
    const currentUserId = useSelector(selectCurrentUserId);

    if (currentUserId) {
        return <></>;
    } else {
        return children;
    }
};

/**
 * Shape of the props that the IsAuthenticated component takes
 */
export declare type IsAuthenticatedProps = {
    render: (auth: boolean) => any | null,
};

/**
 * Component to allow us to conditionally render based on whether we're authenticated or not
 * @param render Render Prop for rendering the child component
 * @constructor
 */
export const IsAuthenticated: FunctionComponent<IsAuthenticatedProps> = ({render}) => {
    const currentUserId = useSelector(selectCurrentUserId);

    return render(currentUserId !== undefined);
};
