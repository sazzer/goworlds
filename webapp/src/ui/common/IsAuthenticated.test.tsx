import configureStore from "redux-mock-store";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import React from "react";
import {Authenticated, IsAuthenticated} from "./Authenticated";

function setup(userId?: string) {
    const mockStoreCreator = configureStore();
    const store = mockStoreCreator({
        currentUserId: {
            userId,
        }
    });

    const element = render(<Provider store={store}><IsAuthenticated render={auth => 'Hello: ' + auth}/></Provider>);

    return {
        store,

        element,
    }
}

it('Renders correctly when authenticated', () => {
    const {element} = setup('userId');

    expect(element.baseElement).toMatchSnapshot();
});

it('Renders correctly when not authenticated', () => {
    const {element} = setup();

    expect(element.baseElement).toMatchSnapshot();
});
