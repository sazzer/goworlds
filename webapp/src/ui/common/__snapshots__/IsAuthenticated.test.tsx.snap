// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when authenticated 1`] = `
<body>
  <div>
    Hello: true
  </div>
</body>
`;

exports[`Renders correctly when not authenticated 1`] = `
<body>
  <div>
    Hello: false
  </div>
</body>
`;
